sonar-application/src/test/java/org/sonar/application/ClusterPropertiesTest.java

```diff
@@ -0,0 +1,92 @@
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package org.sonar.application;
+
+import java.util.Properties;
+import org.junit.Test;
+import org.sonar.process.Props;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.data.MapEntry.entry;
+
+public class ClusterPropertiesTest {
+  @Test
+  public void test_default_values() throws Exception {
+    ClusterProperties props = new ClusterProperties(new Props(new Properties()));
+
+    assertThat(props.getInterfaces())
+      .isEqualTo("");
+    assertThat(props.isPortAutoincrement())
+      .isEqualTo(false);
+    assertThat(props.getPort())
+      .isEqualTo(9003);
+    assertThat(props.isEnabled())
+      .isEqualTo(false);
+    assertThat(props.getMembers())
+      .isEqualTo("");
+    assertThat(props.getName())
+      .isEqualTo("");
+    assertThat(props.getLogLevel())
+      .isEqualTo("WARN");
+  }
+
+  @Test(expected = IllegalStateException.class)
+  public void test_missing_name() {
+    Props props = new Props(new Properties());
+    props.set(ClusterProperties.Parameter.ENABLED.getName(), "true");
+    props.set(ClusterProperties.Parameter.NAME.getName(), "");
+    ClusterProperties clusterProperties = new ClusterProperties(props);
+    clusterProperties.validate();
+  }
+
+  @Test
+  public void test_cluster_with_name() {
+    Props props = new Props(new Properties());
+    props.set(ClusterProperties.Parameter.ENABLED.getName(), "true");
+    props.set(ClusterProperties.Parameter.NAME.getName(), "sonarqube");
+    ClusterProperties clusterProperties = new ClusterProperties(props);
+    clusterProperties.validate();
+  }
+
+  @Test
+  public void test_cluster_properties() {
+    Props props = new Props(new Properties());
+    ClusterProperties clusterProperties = new ClusterProperties(props);
+    clusterProperties.setEnabled(true);
+    clusterProperties.setMembers("192.168.1.1");
+    clusterProperties.setInterfaces("192.168.1.30");
+    clusterProperties.setPort(9003);
+    clusterProperties.setLogLevel("INFO");
+    clusterProperties.setPortAutoincrement(false);
+    clusterProperties.setName("sonarqube");
+    clusterProperties.populateProps(props);
+
+    assertThat(props.rawProperties()).containsOnly(
+      entry(ClusterProperties.Parameter.LOG_LEVEL.getName(), "INFO"),
+      entry(ClusterProperties.Parameter.PORT.getName(), "9003"),
+      entry(ClusterProperties.Parameter.ENABLED.getName(), "true"),
+      entry(ClusterProperties.Parameter.INTERFACES.getName(), "192.168.1.30"),
+      entry(ClusterProperties.Parameter.MEMBERS.getName(), "192.168.1.1"),
+      entry(ClusterProperties.Parameter.NAME.getName(), "sonarqube"),
+      entry(ClusterProperties.Parameter.PORT_AUTOINCREMENT.getName(), "false")
+    );
+  }
+}
```

Comment:
Found the missing property in idea : Settings → Editor → General → Ensure line feed at file end on Save